pkg/chunk/cache/memcached_client.go

```diff
@@ -183,6 +189,11 @@ func (c *memcachedClient) Stop() {
 }
 
 func (c *memcachedClient) Set(item *memcache.Item) error {
+	// Skip hitting memcached at all if the item is bigger than the max allowed size.
+	if c.maxItemSize > 0 && uint64(len(item.Value)) > uint64(c.maxItemSize) {
+		return nil
```

Comment:
Please let me know if you think it would be better to have a counter metric for this like [thanos](https://github.com/thanos-io/thanos/blob/main/pkg/cacheutil/memcached_client.go#L335)